common/src/main/java/io/netty5/util/internal/PlatformDependent0.java

```diff
@@ -223,28 +223,54 @@
             try {
                 final Object maybeDirectBufferConstructorHandle =
                         AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
-                            try {
-                                final Constructor<?> constructor =
+                            if (javaVersion() >= 17) {
+                                try {
+                                    final Constructor<?> constructor =
+                                        direct.getClass().getDeclaredConstructor(long.class, int.class, Object.class,
+                                            java.lang.Class.forName("jdk.internal.access.foreign.MemorySegmentProxy"));
```

Comment:
The type of the last argument will most likely change in Java 19.
Maybe we should instead search the list from `getConstructors()` for a constructor that has 3 or 4 arguments, and the first 3 are `(long, int, Object)`.

Reply:
Done